gum/backend-darwin/gumprocess-darwin.c

```diff
@@ -1306,6 +1307,83 @@ gum_do_enumerate_exports (GumEnumerateExportsContext * ctx,
   return carry_on;
 }
 
+kern_return_t
+gum_malloc_memory_reader (task_t remote_task,
+    vm_address_t remote_address, vm_size_t size, void ** local_memory) {
```

Comment:
I know the current style in frida-gum sucks refactoring-wise, but let's stick to it for now: one argument per line, where the arguments following the first align with it.

Also, internal functions should always be `static`, and should be below the function where it is used, which means a forward-declaration at the top is needed (make sure to insert it chronologically btw). The reason we do it this way is so code always starts from highest level at the top, to lowest level at the bottom, so you can read the source code from the start of the file instead of reading bottom-up.